tools/harness-simulation/harness/Thread_Harness/Sniffer/SimSniffer.py

```diff
@@ -56,7 +60,89 @@
 WINREG_KEY = r'SYSTEM\CurrentControlSet\Control\Network\{4d36e972-e325-11ce-bfc1-08002be10318}'
 
 
+def UIMsgBoxDecorator(UIMsgBox, replaceFuncs):
+
+    @staticmethod
+    def UIMsgBoxWrapper(msg='Confirm ??',
+                        title='User Input Required',
+                        inputRequired=False,
+                        default='',
+                        choices=None,
+                        timeout=None,
+                        isPrompt=False):
+        func = replaceFuncs.get((msg, title))
+        if func is None:
+            UIMsgBox(msg, title, inputRequired, default, choices, timeout, isPrompt)
+        else:
+            func()
+
+    return UIMsgBoxWrapper
+
+
+class DeviceManager:
+
+    def __init__(self):
+        deviceManager = TopologyManager.m_DeviceManager
+        self._devices = {'DUT': deviceManager.AutoDUTDeviceObject.THCI_Object}
+        for device_obj in deviceManager.DeviceList.values():
+            if device_obj.IsDeviceUsed:
+                self._devices[device_obj.DeviceTopologyInfo] = device_obj.THCI_Object
+
+    def __getitem__(self, deviceName):
+        device = self._devices.get(deviceName)
+        if device is None:
+            raise KeyError('Device Name "%s" not found' % deviceName)
+        return device
+
+
+class RFEnclosureManager:
+
+    def __init__(self, deviceNames1, deviceNames2, snifferPartitionId):
+        self.deviceNames1 = deviceNames1
+        self.deviceNames2 = deviceNames2
+        self.snifferPartitionId = snifferPartitionId
+
+        self._reset()
+
+    def _reset(self):
```

Comment:
By the general definition of `reset`, I think we should also clear the deny list for affected nodes and sniffers.

Reply:
Agree. But `_removePartition` (renamed to `_removeRFEnclosure` later) has already cleared the deny list for affected nodes and sniffers. I will change the logic here.